src/osp/Active/SysNewton.cpp

```diff
@@ -19,37 +19,19 @@ void cb_force_torque(const NewtonBody* body, dFloat timestep, int threadIndex)
     auto& transform
             = scene.get_registry().get<ACompTransform>(bodyComp->m_entity);
 
-    // Get new transform matrix from newton
-    Matrix4 matrix;
-    NewtonBodyGetMatrix(body, matrix.data());
-
     // Check if transform is being set
     if (transform.m_transformDirty)
     {
-        // Instead of setting transform directly, it's better to add the
-        // difference with the previous transform. This ensures that the changes
-        // to the rigid body from Newton this frame will still apply.
-
-        // TODO: there's probably a better way to do this. if not, account for
-        //       rotation too, since this is translation only.
+        Matrix4 matrix;
+        NewtonBodyGetMatrix(body, matrix.data());
 
-        Vector3 diffTranslate = transform.m_transform.translation()
-                              - bodyComp->m_prevTransform.translation();
-        matrix.translation() += diffTranslate;
-
-        NewtonBodySetMatrix(body, matrix.data());
+        // Set matrix without Newton 'conveniently' sleeping for a frame
+        NewtonBodySetMatrixNoSleep(body,
+                                   matrix.data());
```

Comment:
What happened to the origin translation? I don't see who is responsible for the transformation now

Reply:
The floating origin stuff was removed from SysNewton, and generalized into setting transform. The translation is now done in SysAreaAssociate, and is requested from DebugCameraController.